### PR TITLE
Build contexts

### PR DESCRIPTION
**Pull request by ferassmohana**

### CODE REVIEW

**Review by ileenshaar — APPROVED** [review]

File: src/components/products-table/ProductsTable.jsx [hunk 1]
```diff
@@ -29,7 +30,7 @@ export const ProductsTable = ({ products, isRemove = true }) => {
         </td>
       </tr>
     ));
-  }, [products]);
+  }, [products, isRemove]);
```
Comment (mahmoud-alragabi) [inline src/components/products-table/ProductsTable.jsx]: rename  ```isRemove```  to  ```showRemoveButton```

File: src/contexts/ProductsContextProvider.jsx
Comment (mahmoud-alragabi) [inline src/contexts/ProductsContextProvider.jsx:1]: rename to ```CartProductsContext```

File: src/contexts/UserContextProvider.jsx
Comment (mahmoud-alragabi) [inline src/contexts/UserContextProvider.jsx:1]: rename variables to users

**Review by mahmoud-alragabi — CHANGES_REQUESTED** [review]
I think it would be better if we had a global context provider that has both the user and cartProducts since we're not using the providers anywhere else but to wrap the router

**Review by mahmoud-alragabi — APPROVED** [review]

**Pull request merged**